mentat/code_context.py

```diff
@@ -345,6 +253,54 @@ def _feature_relative_path(f: CodeFeature) -> str:
 
         return sorted(include_features, key=_feature_relative_path)
 
+    def _get_all_features(
+        self,
+        level: CodeMessageLevel,
+    ) -> list[CodeFeature]:
+        session_context = SESSION_CONTEXT.get()
+        git_root = session_context.git_root
+
+        all_features = list[CodeFeature]()
+        for path in get_non_gitignored_files(git_root):
+            abs_path = git_root / path
+            if (
+                abs_path.is_dir()
+                or not is_file_text_encoded(abs_path)
+                or abs_path in self.ignore_files
+            ):
+                continue
+
+            diff_target = (
+                self.diff_context.target
+                if abs_path in self.diff_context.files
+                else None
+            )
+            user_included = abs_path in self.include_files
+            if level == CodeMessageLevel.INTERVAL:
+                # Return intervals if code_map is enabled, otherwise return the full file
+                full_feature = CodeFeature(
+                    abs_path,
+                    level=CodeMessageLevel.CODE,
+                    diff=diff_target,
+                    user_included=user_included,
+                )
+                if not self.code_map_enabled():
+                    all_features.append(full_feature)
+                else:
+                    _split_features = split_file_into_intervals(
+                        git_root,
+                        full_feature,
+                        user_features=self.include_files.get(abs_path, []),
+                    )
+                    all_features += _split_features
+            else:
+                _feature = CodeFeature(
+                    abs_path, level=level, diff=diff_target, user_included=user_included
+                )
+                all_features.append(_feature)
+
+        return sorted(all_features, key=lambda f: f.path.relative_to(git_root))
```

Comment:
Is `relative_to(git_root)` necessary? Shouldn't they all have the same prefix anyway so it won't affect the sort?